### PR TITLE
Add a mock server for Development and a specialized API Spec

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

* ~Introducing a mock server for development. The idea is to mock all possible requests to the backend so that we **analyze what we need**. The server is implemented using [JSON Server](https://github.com/typicode/json-server) for the following reasons:~
  - ~0 code needed for basic usage, only a JSON schema~
  - ~It allow for more complex behavior such as routes, middleware, authorization..~
  - ~It is well maintained and it is extensively used~
  - ~It is super cool~
  - Not anymore!

* Introducing instead [`open-api-mocker`](https://github.com/jormaechea/open-api-mocker) which is able to interpret an Openapi spec and mock an API. This will allow us both to keep developing (while being backend-agnostic) and to define in a clear way our API needs.

* Also introducing a data access layer (DAL) to abstract from the fetch library, from the route and all the network paraphernalia. We should only worry about getting, updating, etc stuff:
  ```javascript
  const { applications, isLoading } = useGetApplications()
  const { application, isLoading } = useGetApplication({ id: 1 })
  const { application, isLoading } = useUpdateApplication(newApplication)
  ```

* Also introducing a way to organize components: **by page**. General use components should be located under `common/` but that' for another moment to decide.

**Which issue(s) this PR fixes** 

[THREESCALE-4627: Add a mock server to make fake API calls](https://issues.redhat.com/browse/THREESCALE-4627)

### CODE REVIEW

**Comment by didierofrivia** [comment]
I'd rather define it with an OAS3 spec, than a plain json file. Thus we can be more flexible and make many other implementations having the spec.

**Comment by josemigallas** [comment]
@didierofrivia I added a very basic spec.yaml and introduced a new library that will read it and act as a server.

**Comment by damianpm** [comment]
Can you add some mock data, to see something different than 'string' in all fields?
apps

**Comment by josemigallas** [comment]
@damianpm I wouldn't spend time on that since @didierofrivia is creating a new Spec that is supposed to replace this one.

File: portafly/src/tests/pages/Applications.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react'
+
+import { render } from 'tests/setup'
+import { Applications } from 'pages'
+import { useGetApplications } from 'dal/Applications'
+import { getByLabelText } from '@testing-library/react'
```
Comment (damianpm) [inline portafly/src/tests/pages/Applications.test.tsx:6]: `getByLabelText` is never used

File: portafly/package.json [hunk 2]
```diff
@@ -24,15 +24,18 @@
...
+    "date-fns": "^2.9.0",
+    "open-api-mocker": "^1.2.4"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
+    "mock-server": "open-api-mocker -s openapi.yml -p 3001 -w",
     "test": "react-scripts test",
     "test-with-coverage": "yarn test -- --coverage",
     "eject": "react-scripts eject"
```
Comment (damianpm) [inline portafly/package.json:36]: What happened to the "lint" script?

File: portafly/src/components/applications/ApplicationsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { SimpleEmptyState } from 'components'
+import { IApplication } from 'types'
+
+export interface IApplicationsTable {
+  applications: IApplication[]
+}
+
+const ApplicationsTable: React.FunctionComponent<IApplicationsTable> = ({ applications }) => {
+  if (applications.length === 0) {
+    return <SimpleEmptyState msg="There are no Applications" />
```
Comment (damianpm) [inline portafly/src/components/applications/ApplicationsTable.tsx:12]: "There are no Applications" should come from strings file

File: portafly/src/components/EmptyState.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  EmptyStateVariant,
+  EmptyStateIcon,
+} from '@patternfly/react-core'
+import { CubesIcon } from '@patternfly/react-icons'
+
+export interface ISimpleEmptyStateProps {
+  msg: string
+}
+
+const SimpleEmptyState: React.FunctionComponent<ISimpleEmptyStateProps> = ({ msg }) => (
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply portafly/src/components/EmptyState.tsx:14]: Not sure I follow, this `SimpleEmptyState` is shown when a list is empty, the other component `Loading` is for showing a loading state.

**Comment by josemigallas** [comment]
Closing in favor of #1785

**Pull request closed without merging**